rama-tcp/src/client/connect.rs

```diff
@@ -231,37 +237,69 @@ async fn tcp_connect_inner_branch<Dns, Connector>(
     Dns: DnsResolver<Error: Into<BoxError>> + Clone,
     Connector: TcpStreamConnector<Error: Into<BoxError> + Send + 'static> + Clone,
 {
+    
     let ip_it = match ip_kind {
-        IpKind::Ipv4 => match dns.ipv4_lookup(domain).await {
-            Ok(ips) => Either::A(ips.into_iter().map(IpAddr::V4)),
-            Err(err) => {
-                let err = OpaqueError::from_boxed(err.into());
-                tracing::trace!(err = %err, "[{ip_kind:?}] failed to resolve domain to IPv4 addresses");
-                return;
-            }
-        },
-        IpKind::Ipv6 => match dns.ipv6_lookup(domain).await {
+        IpKind::Ipv4 =>{
+            if dns.ipv4_supported(){
```

Comment:
Why do that check here again? You already check if ipv4/ipv6 is supported prior to calling `tcp_connect_inner_branch`